### PR TITLE
Chore/search posts

### PR DESCRIPTION
**Pull request by karljerico**
### OVERVIEW

- Restrict post search by words.
- Added no results found upon searching

### SNIPPETS

image
image
image

### CODE REVIEW

**Review by hipe01mark — DISMISSED** [review]
LGTM 🙇

File: src/app/Services/SearchService.php [hunk 1]
```diff
@@ -12,10 +12,21 @@ class SearchService
...
+
+        foreach ($searchWords as $word) {
+            $posts = Post::where('content', 'like', "% $word %")
+                         ->orWhere('content', 'like', "$word %")
+                         ->orWhere('content', 'like', "% $word")
+                         ->orWhere('content', 'like', $word)
+                         ->get();
+
+            $postResults = array_merge($postResults, $posts->all());
+        }
```
Comment (hipe01mark) [inline src/app/Services/SearchService.php]: What is that HAHAHAH

Comment (hipe01mark) [reply src/app/Services/SearchService.php]: use whereIn instead

Comment (karljerico) [reply src/app/Services/SearchService.php]: Resolved this comment sir. I didn't think that the initial migrate of the contents is in string thats why i cant use REGEX. Thank you sir!

**Review by hipe01mark — APPROVED** [review]
Much better! right?

LGTM 🙇

File: src/app/Services/SearchService.php (hunk 1 above)
Comment (hipe01mark) [reply src/app/Services/SearchService.php]: Based on your previous code, it calls posts every word and that's not good in terms of performance.

File: src/app/Services/SearchService.php [hunk 2]
```diff
@@ -15,7 +15,7 @@ class SearchService
     public function search(string $query): array
     {
         $userResults = User::where('name', 'like', "%$query%")->get();
-        $postResults = Post::where('content', 'like', "%$query%")->get();
+        $postResults = Post::whereRaw("content REGEXP '[[:<:]]" . $query . "[[:>:]]'")->get();
```
Comment (hipe-francis) [inline src/app/Services/SearchService.php:18]: For future references: Be very careful of request queries. Find time to study SQL injection and how to propagate better query builders.

**Review by hipe-francis — APPROVED** [review]
LGTM 🚀

**Pull request merged**